### ISSUE TITLE
continuation-local-storage is still present as a peer dependency on npm

### ISSUE THREAD

**Issue by mrhyde** [request/request-promise#70]
Since you didn't bump package.json version after commit e2d8dfa3b901281d501a507a52a5cf35377cc68a continuation-local-storage  peer dependency is still present in version shipped by npm and causes warnings

**Comment by analog-nico** [comment]
Oh, actually this commit reverts what I added for v1.0.1. But still this peer dependency issue isn't fixed. Do you know a solution?

With the declared peer dependency npm@3 prints:
```
npm WARN EPEERINVALID request-promise@1.0.1 requires a peer of continuation-local-storage@~3 but none was installed.
npm WARN EPEERINVALID cls-bluebird@1.0.1 requires a peer of continuation-local-storage@~3 but none was installed.
```

So I removed the declaration again and npm@3 still prints:
```
npm WARN EPEERINVALID cls-bluebird@1.0.1 requires a peer of continuation-local-storage@~3 but none was installed.
```

The latter was what I wanted to fix in the first place.

cls-bluebird has continuation-local-storage declared as a peer dependency. However, cls-bluebird's code is only required if the user of Request-Promise calls `rp.bindCLS(ns)`. Maybe I could declare cls-bluebird as an optional dependency or so?!

**Comment by mrhyde** [comment]
> Oh, actually this commit reverts what I added for v1.0.1

Yes but since it had no positive impact you should be considering to bump npm version. 
Solutions is to bump version to 1.0.2 and publish it on npm so everyone will stop get this warnings :)

In npm@3 peer dependencies are not installed automatically anymore [link](http://blog.npmjs.org/post/110924823920/npm-weekly-5)

Optional dependency is not a solution ether since npm will try to install it anyway [see](https://docs.npmjs.com/files/package.json#optionaldependencies)

Also i've created PR for cls-bluebird to fix second warning, see https://github.com/TimBeyer/cls-bluebird/pull/4

**Comment by analog-nico** [comment]
I finally took the time to look into the ways to define dependencies and verified that defining cls-bluebird as a regular dependency is the correct way to go.

Thanks for creating the PR for cls-bluebird! IMHO cls-bluebird is right to define continuation-local-storage as a peer dependency since it depends on the namespace interface. Anyway, I guess we are facing a dependency edge case here.

I will bump the version now.

**Comment by analog-nico** [comment]
I just published v1.0.2 on npm. Thanks for your contribution!

**Comment by hildjj** [comment]
Since the CLS support is optional, could you just drop the `cls-bluebird` dependency, and explain in the readme that you need to depend on that and `continuation-local-storage` if you want to use the feature?  I'd rather not take the dependencies in my project since I'm not going to use the feature.

**Comment by mrhyde** [comment]
Note that this kind of changed will require major version bump, so it won't break anyone's code

**Comment by analog-nico** [comment]
Indeed. BTW, we discussed in #75 how to proceed.

**Comment by toboid** [comment]
Also worth noting that in addition to the warning `npm shrinkwrap` fails when using **request-promise** due to the **cls-bluebird** peer dependency on **continuation-local-storage**. Had to work around by adding an explicit dependency on **continuation-local-storage**.

**Comment by analog-nico** [comment]
Hi @toboid I just released `request-promise@2.0.0`. `npm shrinkwrap` works now.

**Comment by toboid** [comment]
great :smile:
